test/CancellationHandleTest.cpp

```diff
@@ -106,3 +116,170 @@ TEST(CancellationHandle, verifyCancellationWorksWithMultipleThreads) {
       CancellationException);
   EXPECT_TRUE(handle.isCancelled());
 }
+
+// _____________________________________________________________________________
+
+TEST(CancellationHandle, ensureObjectLifetimeIsValidWithoutWatchDogStarted) {
+  EXPECT_NO_THROW(CancellationHandle<true>{});
+}
+
+// _____________________________________________________________________________
+
+namespace ad_utility {
+
+TEST(CancellationHandle, verifyWatchDogDoesChangeState) {
+#ifdef __APPLE__
+  GTEST_SKIP_("sleep_for is unreliable for macos builds");
+#endif
+  CancellationHandle<true> handle;
+
+  EXPECT_EQ(handle.cancellationState_, NOT_CANCELLED);
+  handle.startWatchDog();
+
+  // Give thread some time to start
+  std::this_thread::sleep_for(10ms);
+  EXPECT_EQ(handle.cancellationState_, WAITING_FOR_CHECK);
+
+  std::this_thread::sleep_for(detail::DESIRED_CHECK_INTERVAL);
```

Comment:
Shouldn't the sleep be for a little longer to make this test more reliable?

Reply:
Ah, I didn't quite understand the 10ms delay, that should be fine, thanks for clarifying that.